### PR TITLE
Add two more LMTD forms

### PR DESCRIPTION
**Pull request by eslickj**
This starts to bring in stuff from the NGCC branch.  The add two more forms of the LMTD expression.  These may be useful if an approach temperature is near 0. 

### Legal Acknowledgement

By contributing to this software project, I agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the license terms described in the LICENSE.txt file at the top level of this directory.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

File: idaes/generic_models/unit_models/heat_exchanger.py [hunk 1]
```diff
@@ -137,13 +149,64 @@ def delta_temperature_lmtd_callback(b):
...
+    difference (LMTD).  It can be supplied to "delta_temperature_callback"
+    HeatExchanger configuration option. This form is
+
+    .. math::
 
-def delta_temperature_amtd_callback(b):
+        \Delta T = \frac{\Delta T_2 - \Delta T_1}{
+            \log_e\left(\frac{\Delta T_2}{\Delta T_1}\right)}
+
+    where :math:`\Delta T_1` is the temperature differnce at the hot inlet end
```
Comment (andrewlee94) [inline idaes/generic_models/unit_models/heat_exchanger.py:164]: Typo: difference

This one appears a few times.

Comment (eslickj) [reply idaes/generic_models/unit_models/heat_exchanger.py:164]: Dang-old copy paste

**Review by andrewlee94 — APPROVED** [review]
Look OK to me.

**Review by MAZamarripa — APPROVED** [review]
lgtm

**Pull request merged**